src/L1ERC20Bridge.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {ERC20Votes} from "openzeppelin/token/ERC20/extensions/ERC20Votes.sol";
+import {IWormhole} from "wormhole/interfaces/IWormhole.sol";
+
+contract L1ERC20Bridge {
+  /// @notice L1 token used for deposits and voting.
+  ERC20Votes public immutable L1_TOKEN;
+
+  /// @notice Token address which is minted on L2.
+  address public L2_TOKEN_ADDRESS;
+
+  /// @notice The core Wormhole contract used to send messages to L2.
+  IWormhole coreBridge;
+
+  /// @notice A unique number used to send messages.
+  uint32 public nonce;
+
+  /// @notice Used to indicate whether the contract has been intialized with the L2 token address.
+  bool public INITIALIZED = false;
+
+  /// @notice A mapping of users to amount they have deposited into the bridge.
+  mapping(address => uint256) public depositAmount;
```

Comment:
*RR* Why do we track this amount?

Reply:
The idea was for us to use this to validate an L2 withdrawal on L1. The actual logic is handled in a subsequent pr so it isn't necessary in this one.

Reply:
But the address withdrawing the tokens from L2 isn't necessarily the address that bridged them up, right? Because they're transferrable on L2. Or is there something I'm misunderstanding about how this would be used?

Reply:
That is true, yes this can be removed